capstone/fitness/src/main/java/com/google/sps/servlets/UpdateProgressServlet.java

```diff
@@ -0,0 +1,144 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;  
+import com.google.sps.servlets.*;
+import com.google.sps.util.*;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/update-progress")
+public class UpdateProgressServlet extends HttpServlet {
+
+  static Gson gson;  
+  
+  @Override
+  public void init() {
+    gson = new Gson();
```

Comment:
not necessary to add init() just to instantiate gson. You can do inline instantiation when you declare it/